Jumpstart: Adding proper features & style adjustments

jeffgolenski opened the pull request:
Adding the correct list of Jumpstart features and styling them to match the rest of Jetpack React.

To-do:
- [x] Remove the Happiness card at the bottom ( @dereksmart can you write a conditional for this? )
- [x] List of features should be the same as current Jetpack ()
- [x] Make sure we're actually activating the list we're saying we are
- [x] Links on/next to each listed feature to find out more (link to support docs)

(Fixes: https://github.com/Automattic/jetpack/issues/3924)

**Visuals:** 

Before:
Coming soon...

After:

Collapsed
![screen shot 2016-05-25 at 4 40 07 pm](https://cloud.githubusercontent.com/assets/214813/15555405/9cdd0f80-2296-11e6-94c3-cb907ccae222.png)

Expanded:
![screen shot 2016-05-25 at 4 41 26 pm](https://cloud.githubusercontent.com/assets/214813/15555427/b8fd5e04-2296-11e6-97d0-ce3b85cc5e13.png)

Loading:
![screen shot 2016-05-25 at 4 59 10 pm](https://cloud.githubusercontent.com/assets/214813/15555948/343b6f50-2299-11e6-9cf1-fa5bb257aab0.png)


Smaller screens:
![screen shot 2016-05-25 at 4 42 47 pm](https://cloud.githubusercontent.com/assets/214813/15555502/ff482b1e-2296-11e6-8f2a-29a698619c23.png)

jeffgolenski commented:
Only thing left to do is remove the FeedbackDashRequest notice on this particular page.

oskosk commented:
LGTM!

jeffgolenski commented:
Thanks for your help @oskosk !

beaulebens commented on _inc/client/components/jumpstart/index.jsx:
```diff
@@ -29,55 +30,115 @@ const JumpStart = React.createClass( {
+					<FoldableCard
+						className="jp-jumpstart__features"
+						clickableHeaderText={ true }
+						subheader="Learn more"
```
Needs translation wrapper. Also, when I click this, nothing happens -- should it be opening up the rest of the UI or something? I don't see any of the UI in your screenshots, except for the top button basically.

richardmuscat commented:
Looking good @jeffgolenski 

Couple of points:
- why not have the related posts box same width as the others? i know you're going for symmetry but it looks odd
- the spinner seems to be incorrectly aligned to the button
- do you want text edits now or we leave that for another time?

jeffgolenski commented:
@richardmuscat 

> why not have the related posts box same width as the others? i know you're going for symmetry but it looks odd

Why do you think it looks odd? I feel it looks more odd when its not balanced.

> the spinner seems to be incorrectly aligned to the button

Can you provide and screenshot and more details so I can dig into it, puh-lease.

> do you want text edits now or we leave that for another time?

Might as well get them in now!


@beaulebens 

> Needs translation wrapper. Also, when I click this, nothing happens -- should it be opening up the rest of the UI or something? I don't see any of the UI in your screenshots, except for the top button basically.

I've got 4 screenshots that outline its view when it's collapsed and expanded. Did you write this comment before I added those, perhaps? Can you give it another go? The box should slide open when you click learn more. I modified the FoldableCard component to work here - decided that for what it was worth, added a few lines of css to modify that component was much better than creating an entire new custom component to be used just once, here.

beaulebens commented:
For some reason I wasn't seeing the new grid view at all before, but now I am. Carry on :)

The "Learn More" subheader mentioned in my previous comment does need the translation wrapper still though.

richardmuscat commented:
@jeffgolenski 

> Why do you think it looks odd? I feel it looks more odd when its not balanced.

I think its odd because its putting form over function. While the wider box balances the boxes, it makes the related posts content harder to read and scan. There's no additional content to justify the bigger box.

> Can you provide and screenshot and more details so I can dig into it, puh-lease.

I'm just going off the [screenshot](https://cloud.githubusercontent.com/assets/214813/15555948/343b6f50-2299-11e6-9cf1-fa5bb257aab0.png) you posted yourself up top titled "Loading".

> Text (and @jessefriedman I'd love your thoughts here)
- I don't think we need to Learn More links at all here
- I was originally going to suggest changing the button text to "Jump Start" but I actually like the new text now

jessefriedman commented:
@jeffgolenski regarding the balance issue I agree with @richardmuscat that it shouldn't be full width if we have an odd number of features. That being said I think I have a solution. Not sure why sharing and related posts got removed from this list, (maybe I missed the conversation around that?), but I'd like to see them added back, along with Monitor. 

If we add Related Posts, Sharing and Monitor it evens things out and you won't have an odd number of features in the list. 

_____

As for the "learn more" text. I agree that it's not a required piece. We can watch if people use Jump Start less and go from there. My only thing is that Jump Start affects the front end of your site, especially if we go with Related Posts and Sharing like I suggested above. This is another thing we can watch and see if people are quickly deactivating those features after they Jump Start or if tickets come in about it. 

_____ 

^ Leads me to my next point, I think we should auto Jump Start sites that are very young. If your site is less than a few days old it makes more sense to me that these features are just on by default. @richardmuscat  we had a discussion around this last year. 


_____

Content:

Instead of 

> "Quickly enhance your site by activating Jetpack's recommended features"

how about:

> "Enable Jetpack's recommended features for improved site performance, security, and visitor engagement. "

richardmuscat commented:
@jessefriedman: related posts is already in there!

jessefriedman commented:
@richardmuscat I mentioned Related Posts initially but quickly edited.  If we add Sharing back it will balance everything.  If we go with Monitor as well it will unbalance everything but it's not the end of the world.

richardmuscat commented:
I just disagree with the principle of adding or removing a feature for visual balance. Form over function again :-)

If we have five features let's just show five :-)

> On 2 Jun 2016, at 11:59, Jesse Friedman <notifications@github.com> wrote:
> 
> @richardmuscat I mentioned Related Posts initially but quickly edited. If we add Sharing back it will balance everything. If we go with Monitor as well it will unbalance everything but it's not the end of the world.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

jessefriedman commented:
I disagree with adding a feature for balance, but Sharing belongs in that list. I'm not sure how it got removed or why? I don't think there was a discussion about it, I think it just got removed from a previous version of Jetpack by accident.  I also believe that adding Monitor is the right move. It's a very popular feature and the most popular "first move" users make after jump start.

richardmuscat commented:
Ok. IMO this is not the time to change what Jump Start does. So we need a
canonical list of what is currently in Jump Start and stick with that. If
Sharing is in, then its in. If not, then its out :-)

On Thu, Jun 2, 2016 at 12:10 PM, Jesse Friedman <notifications@github.com>
wrote:

> I disagree with adding a feature for balance, but Sharing belongs in that
> list. I'm not sure how it got removed or why? I don't think there was a
> discussion about it, I think it just got removed from a previous version of
> Jetpack by accident. I also believe that adding Monitor is the right move.
> It's a very popular feature and the most popular "first move" users make
> after jump start.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Automattic/jetpack/pull/3967#issuecomment-223261988>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACLofJ4eyLc2T7LvXZK_TdTivOJT2MSzks5qHro0gaJpZM4Im0NX>
> .
>

jessefriedman commented:
> IMO this is not the time to change what Jump Start does

Cool so we won't do Monitor

> So we need a canonical list of what is currently in Jump Start and stick with that. 

Photon, Manage, Single Sign On, Related Posts, Carousel, Sharing

That's what the list is supposed to be. Not sure when sharing got removed but it definitely belongs in the list.

dereksmart commented:
@jessefriedman @jeffgolenski @richardmuscat Would it be helpful if I propagated that list programmatically based on what is in the module headers?  This is also what controls what is actually activated on Jump Start.

jessefriedman commented:
if it's what is controlling what actually gets jump started then that list
should be under the same control, so yes and thank you
On Thu, Jun 2, 2016 at 1:12 PM Derek Smart <notifications@github.com> wrote:

> @jessefriedman <https://github.com/jessefriedman> @jeffgolenski
> <https://github.com/jeffgolenski> @richardmuscat
> <https://github.com/richardmuscat> Would it be helpful if I propagated
> that list programmatically based on what is in the module headers? This is
> also what controls what is actually activated on Jump Start.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Automattic/jetpack/pull/3967#issuecomment-223358198>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABqz4sklZ5hSyCihDQyN3szF_ja0zkK-ks5qHw8DgaJpZM4Im0NX>
> .
>

dereksmart commented:
@jessefriedman great, I will push to this branch.  

cc @jeffgolenski

dereksmart commented:
@richardmuscat @jessefriedman @jeffgolenski Everything is dynamic now.  Pulling in the data from the module headers as it is in current Jetpack.  All modules you see listed will also be jump started.  

FYI doing this broke the 2-col layout, so it's all in one-column ATM

dereksmart commented:
This is now ready for review.  I've made a couple of adjustments so that it's now 2-col and the last (odd) module does not span full-width

cc @jeffgolenski @richardmuscat 

![jumpstart-modules](https://cloud.githubusercontent.com/assets/7129409/15793577/eca69b58-29b2-11e6-89c1-637b60d4aeb6.png)

richardmuscat commented:
@dereksmart to me that looks good. While I would like to reconsider what goes into Jumpstart I think that should be a separate discussion to this.

My only gripe is removing those "Learn more" links from the boxes. I feel for the purposes of Jumpstart the short descriptions should be enough.

cc @jeffgolenski @jessefriedman

dereksmart commented:
@richardmuscat I've removed the learn more links.  This PR should be good to go @jeffgolenski

jeffgolenski commented:
IPF
